### PR TITLE
Lint all files

### PR DESCRIPTION
**Pull request by OscarB28**
Linted all files
Also fixed a bug with the timezones causing a problem on AWS

### CODE REVIEW

File: pipeline/extract.py [hunk 1]
```diff
@@ -33,18 +33,18 @@ def get_number_of_plants(api_index: str):
         # and go to 51 since loops aren't inclusive on the last number
         return number_of_plants + LOOP_INCLUSIVE_NUMBER
 
-    except KeyError:
+    except KeyError as exc:
```
Comment (IB64) [inline pipeline/extract.py:36]: Nice way of re-raising an exception

File: pipeline/extract.py [hunk 2]
```diff
@@ -68,29 +68,29 @@ def connect_to_plant_ids(total_num_plants: int, api_plants: str):
...
-    Converts the list of all plant dictionaries into a pandas dataframe.
+    Converts the list of all plant dictionaries into a pandas DataFrame.
     """
 
-    df = pd.DataFrame(plants)
-    df = df.fillna("N/A")
-    return df
+    plants_df = pd.DataFrame(all_plant_data)
+    plants_df = plants_df.fillna("N/A")
+    return plants_df
```
Comment (caithomas6789) [inline pipeline/extract.py:93]: Dataframe name is a lot more clear now.

File: pipeline/load.py [hunk 3]
```diff
@@ -21,43 +25,45 @@ def get_database_connection():
 
         return engine.connect()
 
-    except ConnectionError as error:
-        print(error)
-    pass
+    except ConnectionError as exc:
+        raise DBConnectionError(
+            {'error': True, 'message': 'Connection Failed'}, BAD_REQUEST) from exc
```
Comment (caithomas6789) [inline pipeline/load.py:30]: Good use of the custom Database error.

File: pipeline/load.py [hunk 4]
```diff
@@ -21,43 +25,45 @@ def get_database_connection():
...
-    conn.execute(sql.text("COMMIT;"))
-    res = conn.execute(query).fetchall()
+    db_conn.execute(sql.text("COMMIT;"))
+    res = db_conn.execute(query).fetchall()
 
     return res
 
 
-def insert_data_into_database(conn, rows):
+def insert_data_into_database(db_conn, df_rows):
```
Comment (caithomas6789) [inline pipeline/load.py:48]: Clear variable names.

File: pipeline/test_transform.py [hunk 5]
```diff
@@ -17,14 +17,15 @@ def test_last_watered_all_datetime():
...
+                   "soil_moisture": [33.243289352, 29.234892343]}
     df = pd.DataFrame(data=filled_data).reset_index(drop=True)
 
     df = clean_last_watered_data(df)
 
     dict_df = df.to_dict()
-    assert dict_df == {"last_watered": {0: Timestamp("2023-12-18 13:54:32+0000", tz="GMT"),
-                                        1: Timestamp("2023-12-19 09:54:32+0000", tz="GMT")},
+    assert dict_df == {"last_watered": {0: Timestamp("2023-12-18 13:54:32"),
+                                        1: Timestamp("2023-12-19 09:54:32")},
```
Comment (caithomas6789) [inline pipeline/test_transform.py:28]: Good fix to the issue.

File: pipeline/transform.py [hunk 6]
```diff
@@ -1,72 +1,84 @@
 """
 Transform script for the plants pipeline.
 """
-import pandas as pd
+
 from time import perf_counter
+import pandas as pd
```
Comment (caithomas6789) [inline pipeline/transform.py:6]: Keeping the imports in order is good practice

File: pipeline/transform.py [hunk 7]
```diff
@@ -1,72 +1,84 @@
...
+    plants_df = plants_df.drop(
+        plants_df[plants_df.temperature < MINIMUM_TEMPERATURE].index)
+    plants_df = plants_df.drop(
+        plants_df[plants_df.temperature > MAXIMUM_TEMPERATURE].index)
 
-    df = df.dropna()
+    plants_df = plants_df.dropna()
 
-    return df
+    return plants_df
```
Comment (caithomas6789) [inline pipeline/transform.py:81]: Dataframe name is much clearer now.

File: rds_s3_pipeline/test_daily_extract.py [hunk 8]
```diff
@@ -20,27 +20,40 @@ def test_extract_from_rds_works():
...
     assert mock_execute.call_count == 2
     assert mock_fetch.call_count == 1
-    assert results == [(2, datetime.datetime(2023, 12, 19, 11, 0, 45), 25.22258419783465, 12.013140983634887, datetime.datetime(
-        2023, 12, 18, 13, 54, 32), 1), (3, datetime.datetime(2023, 12, 19, 11, 0, 46), 31.762604085532303, 9.08111140376697, datetime.datetime(2023, 12, 18, 14, 10, 54), 2)]
+    assert results == [(2, datetime.datetime(2023, 12, 19, 11, 0, 45),
+                        25.22258419783465, 12.013140983634887,
+                        datetime.datetime(2023, 12, 18, 13, 54, 32), 1),
+                       (3, datetime.datetime(2023, 12, 19, 11, 0, 46),
+                        31.762604085532303, 9.08111140376697,
+                        datetime.datetime(2023, 12, 18, 14, 10, 54), 2)]
```
Comment (caithomas6789) [inline rds_s3_pipeline/test_daily_extract.py:39]: Looks a lot neater this way.

**Review by caithomas6789 — APPROVED** [review]

**Pull request merged**